trieste/objectives/multi_objectives.py

```diff
@@ -236,3 +248,66 @@ def gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
         search_space=Box([0.0], [1.0]) ** d,
         gen_pareto_optimal_points=gen_pareto_optimal_points,
     )
+
+
+def ConstrainedBraninCurrin() -> ConstrainedMultiObjectiveTestProblem:
+    """
+    The ConstrainedBraninCurrin problem, typically evaluated over :math:`[0, 1]^2`.
+    See :cite:`belakaria2019max` and :cite:`daulton2020differentiable`
+    (the latter for adding the constraint) for details.
+
+    :return: The problem specification.
+    """
+
+    def gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
+        return tf.zeros(shape=0, dtype=tf.float64)
+
+    def evaluate_constraint(x: TensorType, threshold: Optional[float] = 0.0) -> TensorType:
```

Comment:
I think we should have some tests for this, for example:
1. for some selected input and threshold values we know what the result should be
2. the output is boolean

Reply:
not sure if get point 1 properly but we have a test for different `threshold values` as [here](https://github.com/TsingQAQ/trieste/blob/1ae93d133b019ab4e3a54d3b4fe8c6896fa7a313/tests/unit/objectives/test_multi_objectives.py#L133-L142), would it be possible to elaborate these points especially point 2 please?

Reply:
indeed, you have covered 1. already, I have missed that, apologies

regarding 2. its quite simple, just add a test that the constraint output has correct dtype, e.g. you can modify `test_objective_and_constraint_has_correct_shape_and_dtype` by adding a check on constraint dtype if constraint is present